### ISSUE TITLE
Caching test for bearer token authentication intermittently fails

### ISSUE THREAD

**Issue by rehandalal** [mozilla/normandy#1423]
The test:
`normandy.base.tests.api.test_authentication.TestBearerTokenAuthentication.test_caching`

We currently use `sleep` to add a delay that is 1 second longer than the time in which the cache should expire but this appears to intermittently cause issues and can hopefully be done in a better way.

**Comment by peterbe** [comment]
Is it just that tests that fails intermittently?

**Comment by mythmon** [comment]
@peterbe I'm not sure what you're asking. Can you rephrase your question?

**Comment by peterbe** [comment]
It is just that one test `TestBearerTokenAuthentication.test_caching` or are there other tests as well?

Actually, when I re-read the description it does say "The test:" so it's probably only that one single test.

**Comment by mythmon** [comment]
Ah, yes. It is just the one test that is failing intermittently, from what I've seen.

**Comment by peterbe** [comment]
I encountered this too so I thought I'd note the pytest signature. 

<details>
_____________________________________ TestBearerTokenAuthentication.test_caching _____________________________________

self = <normandy.base.tests.api.test_authentication.TestBearerTokenAuthentication object at 0x7f7843c85a58>
csrf_api_client = <rest_framework.test.APIClient object at 0x7f7842e03ba8>
requestsmock = <requests_mock.mocker.Mocker object at 0x7f7842e03c18>
settings = <pytest_django.fixtures.SettingsWrapper object at 0x7f784431e470>

    def test_caching(self, csrf_api_client, requestsmock, settings):
        user = UserFactory()
        user_data = json.dumps(
            {"email": user.email, "given_name": user.first_name, "family_name": user.last_name}
        ).encode("utf-8")

        ratelimit_reset = int(time.mktime(datetime.utcnow().timetuple())) + 1

        requestsmock.get(
            settings.OIDC_USER_ENDPOINT,
            [
                {
                    "content": user_data,
                    "status_code": 200,
                    "headers": {"X-RateLimit-Reset": f"{ratelimit_reset}"},
                },
                {"status_code": 401},
            ],
        )

        response = csrf_api_client.post(
            "/bearer/", {"example": "example"}, HTTP_AUTHORIZATION="Bearer valid-token"
        )
        assert response.status_code == 200
        assert response.data.get("user") == user.email

        # Response should be cached and you shouldn't hit the 401
        response = csrf_api_client.post(
            "/bearer/", {"example": "example"}, HTTP_AUTHORIZATION="Bearer valid-token"
        )
        assert response.status_code == 200
        assert response.data.get("user") == user.email

        # Sleep till cache expires
        time.sleep(2)
        response = csrf_api_client.post(
            "/bearer/", {"example": "example"}, HTTP_AUTHORIZATION="Bearer valid-token"
        )
>       assert response.status_code == 401
E       assert 200 == 401
E        +  where 200 = <Response status_code=200, "application/json">.status_code
</details>

**Comment by peterbe** [comment]
[This just hit us on `master` in Circle](https://circleci.com/gh/mozilla/normandy/4672?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) I'm going to try to fix it.
